What’s up in Vietnam.md

```diff
@@ -107,19 +108,19 @@ _Ted Neward_
+7. ORM frameworks using Active Record pattern incentivize bad practices, like mixing persistence layer with the domain layer, creating anemic domain objects and breaking the Single Responsibility Principle. Trained software architects can avoid this and retain the sharp separation between persistence and domain model in exchange for painful maintenance.
 
 ## Summary
-As we can see, Vietnam is slowly changing, or to be more precise: it adjusts itself to the external world. Is it enough to deal with it in current form? It depends. I’m not against ORM frameworks in general. There are places where we’re operating on relational data which can be mapped merely to object-oriented world. Especially in small applications with domination of composition in the domain layer. What to do in other situations? Get to know and consider all limitations of usage of ORM framework and deal with it … or find other solution.
+As we can see, Vietnam is slowly changing, or to be more precise: it adjusts itself to the external world. Is it enough to deal with it in its current form? It depends. I’m not against ORM frameworks in general. There are places where we’re operating on relational data which can be mapped simply to the object-oriented world. Especially in small applications with domination of composition in the domain layer. What to do in other situations? Get to know and consider all limitations an ORM framework imposes and deal with it … or find a different solution.
 
-1. Use a proper database to your need. Maybe it’s better to use document storage/key-value store (like in the example with smart devices), graph database (in many social services) or other types of database.
+1. Use a proper database to your need. Maybe it’s better to use a document storage/key-value store (like in the example with smart devices), a graph database (in many social services) or a completely different type of database.
 2. Use CQRS, save events as known truths and visualize data in reporting stores in proper databases: relational, context-dependent data in RDBS, objects in key-value stores, graphs in graph databases etc.
-3. Use half-way solution shying away from ORM frameworks, but still using RDBS. It’s recommended approach if we want to keep the clear separation between domain and persistence layer without additional maintenance of persistence model needed for ORM framework. In exchange, we can use the database with plain SQL more freely without the risk of side effects.
+3. Use a half-way solution shying away from ORM frameworks, but still using RDBS. It’s a recommended approach if we want to keep a clear separation between domain and persistence layer without additional maintenance A persistence model needed for ORM framework would bring. In exchange, we can use the database with plain SQL more freely without the risk of causing side effects.

```

Comment:
You sure about this capital A in the middle?

Reply:
I'm sure it's wrong :d 'of a' is what  I was looking for I think.